### PR TITLE
SageMaker types changes

### PR DESCRIPTION
**Pull request by EngHabu**
# TL;DR
- [x] Use structs to unmarshal hyperparameterconfig,
- [x] Read parameter ranges from inputs

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Tracking Issue
https://github.com/lyft/flyte/issues/455

### CODE REVIEW

**Comment by EngHabu** [comment]
ping @bnsblue @kumare3

**Comment by kumare3** [comment]
Will loook tomorrow morning

File: go/tasks/plugins/k8s/sagemaker/utils.go [hunk 1]
```diff
@@ -122,75 +125,102 @@ func getPrebuiltTrainingImage(ctx context.Context, job *flyteSagemakerIdl.Traini
...
 	var retValue = &commonv1.ParameterRanges{
 		CategoricalParameterRanges: []commonv1.CategoricalParameterRange{},
 		ContinuousParameterRanges:  []commonv1.ContinuousParameterRange{},
 		IntegerParameterRanges:     []commonv1.IntegerParameterRange{},
 	}
 
-	for prName, pr := range prMap {
-		scalingTypeString := strings.Title(strings.ToLower(pr.GetContinuousParameterRange().GetScalingType().String()))
-		switch pr.GetParameterRangeType().(type) {
+	errs := errors.ErrorCollection{}
```
Comment (bnsblue) [inline go/tasks/plugins/k8s/sagemaker/utils.go:142]: This is not used except in the `return` statement.

**Comment by bnsblue** [comment]
I am trying to run the propeller with this plugin locally not am not able to for some reasons 
Seeing this error msg:
```
{"json":{"src":"root.go:246"},"level":"fatal","msg":"Error running controller: failed to load plugin - sagemaker_hyperparameter_tuning: no matches for kind \"HyperparameterTuningJob\" in version \"sagemaker.aws.amazon.com/v1\"","ts":"2020-10-14T00:39:45-07:00"}
```
Might be my bad but I am not sure what went wrong

**Comment by bnsblue** [comment]
Trying to run end2end in a sandbox to go through some details. Will continue tomorrow morning.

**Comment by bnsblue** [comment]
@EngHabu  I ran the [test workflow](https://github.com/lyft/flytekit/blob/c48d4969301d123cd901fd09f668130bdc0d7b7c/tests/flytekit/common/workflows/sagemaker.py) you put inside your `flytekit` PR locally and was stepping through the code with it. I noticed that the input `static_hyperparameters`, which is NOT a `ParameterRangeOneOf`, can get past this UnmarshalStruct check (err == nil). 
https://github.com/lyft/flyteplugins/pull/123/files#diff-0860ebd6bc768e12560fd1608c1053f3c78899a4c3028e82e17b70f59d86ffe5R148-R149

Is this as expected?

**Comment by bnsblue** [comment]
Otherwise, this PR LGTM.

**Comment by EngHabu** [comment]
Excellent point. What will happen in that scenario is that the we will skip this input. (won't be considered a tunable hyperparameter)... That's why I added this TODO: https://github.com/lyft/flyteplugins/pull/123/files#diff-0860ebd6bc768e12560fd1608c1053f3c78899a4c3028e82e17b70f59d86ffe5R188-R190
We can't reliably throw an error unless we can inspect the interface and recognize certain inputs as ParameterRanges... this is a bit tricky today because we use Generic/Proto to represent those inputs...

**Review by bnsblue — APPROVED** [review]

**Pull request merged**